lib/msal-browser/src/cache/AsyncMemoryStorage.ts

```diff
@@ -112,12 +116,26 @@ export class AsyncMemoryStorage<T> implements IAsyncStorage<T> {
     /**
      * Clears in-memory Map and tries to delete the IndexedDB database.
      */
-    async clear(): Promise<void> {
+    async clear(): Promise<boolean> {
+    // InMemory cache is a Map instance, clear is straightforward
+        this.logger.verbose(`Deleting in-memory keystore ${this.storeName}`);
         this.inMemoryCache.clear();
+        this.logger.verbose(`In-memory keystore ${this.storeName} deleted`);
+        this.logger.verbose(`Deleting persistent keystore ${this.storeName}`);
+        
         try {
-            await this.indexedDBCache.deleteDatabase();
+            const dbDeleted = await this.indexedDBCache.deleteDatabase();
+            
+            if (dbDeleted) {
+                this.logger.verbose(`Persistent keystore ${this.storeName} deleted`);
+            }
+            
+            return dbDeleted;
         } catch (e) {
-            this.handleDatabaseAccessError(e);
+            if (e instanceof Error) {
```

Comment:
Why is this check needed? Can you add a comment explaining this?

Reply:
It wasn't needed, I think I added it during debugging and didn't remove it.